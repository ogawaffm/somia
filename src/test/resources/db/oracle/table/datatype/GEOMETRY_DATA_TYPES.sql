BEGIN
    EXECUTE IMMEDIATE 'DROP TABLE TEST.GEOMETRY_DATA_TYPES';
EXCEPTION
    WHEN OTHERS THEN
        IF SQLCODE != -942 THEN
            RAISE;
        END IF;
END;
/
CREATE TABLE TEST.GEOMETRY_DATA_TYPES
(
    SDO_GEOMETRY_COL             SDO_GEOMETRY,
    SDO_TOPO_GEOMETRY_COL        SDO_TOPO_GEOMETRY,

    SDO_GEOR_HISTOGRAM_COL       SDO_GEOR_HISTOGRAM,
    SDO_GEOR_HISTOGRAM_ARRAY_COL SDO_GEOR_HISTOGRAM_ARRAY,

    SDO_GEOR_COLORMAP_COL        SDO_GEOR_COLORMAP,
    --SDO_GEOR_GRAYSCALE_COL       SDO_GEOR_GRAYSCALE,

    --SDO_RASTERSET_COL            SDO_RASTERSET,

    SDO_RANGE_COL                SDO_RANGE,

    SDO_RANGE_ARRAY_COL          SDO_RANGE_ARRAY,
    SDO_RASTER_COL               SDO_RASTER,
    SDO_GEOR_SRS_COL             SDO_GEOR_SRS,

    SDO_GEOR_GCP_COL             SDO_GEOR_GCP,
    SDO_GEOR_GCPGEOREFTYPE_COL   SDO_GEOR_GCPGEOREFTYPE,
    SDO_GEOR_GCP_COLLECTION_COL  SDO_GEOR_GCP_COLLECTION

)