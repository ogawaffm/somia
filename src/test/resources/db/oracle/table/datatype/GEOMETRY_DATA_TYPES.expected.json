{
  "name": "GEOMETRY_DATA_TYPES",
  "catalogName": "",
  "schemaName": "TEST",
  "typeName": "TABLE",
  "comment": "",
  "columns": [
    {
      "name": "SDO_GEOMETRY_COL",
      "type": {
        "name": "SDO_GEOMETRY",
        "typeNumber": 1111,
        "className": "java.lang.Object",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "SDO_TOPO_GEOMETRY_COL",
      "type": {
        "name": "SDO_TOPO_GEOMETRY",
        "typeNumber": 1111,
        "className": "java.lang.Object",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "SDO_GEOR_HISTOGRAM_COL",
      "type": {
        "name": "SDO_GEOR_HISTOGRAM",
        "typeNumber": 1111,
        "className": "java.lang.Object",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "SDO_GEOR_HISTOGRAM_ARRAY_COL",
      "type": {
        "name": "SDO_GEOR_HISTOGRAM_ARRAY",
        "typeNumber": 1111,
        "className": "java.lang.Object",
        "precision": 3752,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "SDO_GEOR_COLORMAP_COL",
      "type": {
        "name": "SDO_GEOR_COLORMAP",
        "typeNumber": 1111,
        "className": "java.lang.Object",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "SDO_RANGE_COL",
      "type": {
        "name": "SDO_RANGE",
        "typeNumber": 1111,
        "className": "java.lang.Object",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "SDO_RANGE_ARRAY_COL",
      "type": {
        "name": "SDO_RANGE_ARRAY",
        "typeNumber": 1111,
        "className": "java.lang.Object",
        "precision": 3752,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "SDO_RASTER_COL",
      "type": {
        "name": "SDO_RASTER",
        "typeNumber": 1111,
        "className": "java.lang.Object",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "SDO_GEOR_SRS_COL",
      "type": {
        "name": "SDO_GEOR_SRS",
        "typeNumber": 1111,
        "className": "java.lang.Object",
        "precision": 3,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "SDO_GEOR_GCP_COL",
      "type": {
        "name": "SDO_GEOR_GCP",
        "typeNumber": 1111,
        "className": "java.lang.Object",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "SDO_GEOR_GCPGEOREFTYPE_COL",
      "type": {
        "name": "SDO_GEOR_GCPGEOREFTYPE",
        "typeNumber": 1111,
        "className": "java.lang.Object",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "SDO_GEOR_GCP_COLLECTION_COL",
      "type": {
        "name": "SDO_GEOR_GCP_COLLECTION",
        "typeNumber": 1111,
        "className": "java.lang.Object",
        "precision": 3752,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    }
  ],
  "indexes": [
    {
      "name": "",
      "columns": [],
      "isUnique": true,
      "comment": "",
      "qualifierName": "",
      "isClustered": false,
      "isHashed": false,
      "isStatistic": true,
      "whereClause": "",
      "cardinality": 0,
      "pages": 0
    }
  ],
  "collation": "",
  "primaryKey": null,
  "bestRowIdentifier": null
}